app/services/hyrax/valkyrie_persist_derivatives.rb

```diff
@@ -25,11 +25,13 @@ def self.call(stream,
       # responds to #path -- here we only have a StringIO, so some
       # transformation is in order
       tmpfile = Tempfile.new(file_set.id, encoding: 'ascii-8bit')
-      tmpfile.write stream.read
+      stream.rewind
+      output = tmpfile.write(stream.read)
+      tmpfile.flush
```

Comment:
the adapter does not flush the buffer first, so we need to make absolutely sure the file is in the tmp location.